### PR TITLE
feat: 🎸 Add `Subsidy` entity 

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### Changelog

* Create a new `Subsidy` entity with following functionality - 
    * `getAllowance` - Provides the current allowance for the Subsidy relationship
    * `quit` - Procedure that can be called by the subsidizer or the beneficiary to leave the Subsidy relationship

BREAKING CHANGE: 🧨 `Account.getSubsidy` now returns `SubsidyWithAllowance | null`

### CODE REVIEW

File: src/api/entities/Subsidy/index.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { Account, Context, Entity, quitSubsidy } from '~/internal';
+import { NoArgsProcedureMethod } from '~/types';
+import { createProcedureMethod, toHumanReadable } from '~/utils/internal';
+
+export interface UniqueIdentifiers {
+  beneficiaryAddress: string;
+  subsidizerAddress: string;
```
Comment (monitz87) [inline src/api/entities/Subsidy/index.ts:7]: I think `beneficiary` and `subsidizer` are enough here. We can explain they're addresses with JSDoc comments ("beneficiary address", "subsidizer address")

File: src/api/entities/Subsidy/index.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  public async exists(): Promise<boolean> {
+    const {
+      beneficiary: { address: beneficiaryAddress },
+      subsidizer: { address: subsidizerAddress },
+      context,
+    } = this;
+
+    const subsidy = await context.accountSubsidy(beneficiaryAddress);
+
+    return subsidy !== null && subsidy.subsidizer.address === subsidizerAddress;
```
Comment (monitz87) [inline src/api/entities/Subsidy/index.ts:74]: You should be able to use `subsidy.subsidizer.isEqual(subsidizer)`

**Comment by monitz87** [comment]
You should modify `Context.accountSubsidy` and `Account.getSubsidy` to return

```
interface SubsidyWithAllowance {
  subsidy: Subsidy;
  allowance: BigNumber;
}
```

And also add `getAllowance(): Promise<BigNumber>` to the `Subsidy` entity

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/quitSubsidy.ts [hunk 3]
```diff
@@ -33,6 +33,8 @@ export async function prepareQuitSubsidy(
 
   const exists = await subsidy.exists();
 
+  console.log(exists);
```
Comment (monitz87) [inline src/api/procedures/quitSubsidy.ts:36]:
```suggestion
```

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**